c_emulator/riscv_sim.c

```diff
@@ -1088,15 +1093,29 @@ int main(int argc, char **argv)
     }
     printf("Connected\n");
   } else
-    entry = load_sail(file);
+    entry = load_sail(initial_elf_file);
 #else
-  uint64_t entry = load_sail(file);
+  uint64_t entry = load_sail(initial_elf_file);
 #endif
+  /* Load any additional ELF files into memory */
+  for (int i = files_start + 1; i < argc; i++) {
+    bool is32bit;
+    uint64_t additional_entry;
+    fprintf(stdout, "Loading additional ELF file %s.\n", argv[i]);
+    /*
+     * We use load_elf() directly instead of load_sail() as we do not want to
+     * override tohost/{begin,end}_signature from the initial ELF file.
```

Comment:
Why not add a bool to load_sail that makes it early return after the loading?

Reply:
Done. Thanks for the suggestion, makes it cleaner.